src/layout.jl

```diff
@@ -119,6 +119,10 @@ function spring_layout(g::AbstractGraph,
     force_x = zeros(nvg)
     force_y = zeros(nvg)
 
+    # Convert locs to float
+    locs_x = float.(locs_x_in)
+    locs_y = float.(locs_y_in)
```

Comment:
```suggestion
    locs_x = convert(Vector{Float64, locs_x_in})
    locs_y = convert(Vector{Float64, locs_y_in))
```
This way, we do  not allocate a new array, if the inputs are already of type `Vector{Float64}`.

Reply:
I didn't know this. Great. thanks.